### ISSUE TITLE
some issue with distance to rooms not being counted from all owned rooms?

### ISSUE THREAD

**Issue by Somotaw** [TooAngel/screeps#102]
[22:36:01]16591357 W29N78 scoutnextroom-819    [room W29N78 pos 8,49] claim

I own room W29N77 and I set config.nextRoom.minNewRoomDistance = 2;

**Comment by TooAngel** [comment]
I had the issue that only single source rooms were claimed. My gut feeling is that the creeps enters a hostile controlled room, but sets somehow the 'memory.claimRoom'. But moves back to the previous room and starts claiming.
To validate the thesis we could set claiming to the roomName instead of true and check.

**Comment by Somotaw** [comment]
I will check this, but im also suggesting making the room check if there are any rooms at minNewRoomDistance+value accessible and test those for claim before spawning a scoutnextroom. Still need to fix this though :)

**Comment by TooAngel** [comment]
TL;DR I like the idea, much better than sending out scoutnextroom from each room.

I think this goes hand in hand with https://github.com/TooAngel/screeps/issues/88#issuecomment-270766474

So store in memory (room cache) a list of source IDs. In combination with the room state we can decide for a good new room to claim: `_filter(Memory.rooms, {return 2 sources && state.in(reservered, unreserved) && far away from existing room}.

Later on we could even rate the rooms based on the mineral available.

**Comment by Somotaw** [comment]
Yes. Combining the two sounds perfect. We can have the regular scouting do the groundwork :)

**Comment by Somotaw** [comment]
hehe just need to fix regular scouting first... ;)
